### PR TITLE
feat: [SRM-11393]: GCO metric template

### PR DESCRIPTION
**Pull request by deepesh-ui**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 2 Issues 
* Code issue 2 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10618)

**Review by praneshg239 — APPROVED** [review]

File: src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.tsx [hunk 1]
```diff
@@ -312,20 +316,24 @@ export function GCOMetricsHealthSource(props: GCOMetricsHealthSourceProps): JSX.
...
-                        onClickExpand={setIsQueryExpanded}
-                        isDialogOpen={isQueryExpanded}
-                        query={formikProps.values.query}
-                        loading={loading}
-                        textAreaName={FieldNames.QUERY}
-                      />
-
+                      <Container width={'500px'}>
+                        <QueryContent
+                          handleFetchRecords={async () => {
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.tsx:321]: why `async` here?

File: src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.utils.ts [hunk 2]
```diff
@@ -74,6 +75,12 @@ export const getSelectedDashboards = (sourceData: any) => {
   return selectedDashboards
 }
 
+// type StackdriverMetricDefinition = StackdriverMetricHealthSourceSpec['metricDefinitions']
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.utils.ts:78]: please remove commented code

File: src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/components/SelectGCODashboards/SelectGCODashboards.tsx [hunk 3]
```diff
@@ -13,7 +13,12 @@ import MetricsDashboardList from '@cv/components/MetricsDashboardList/MetricsDas
 import type { TableDashboardItem } from '@cv/components/MetricsDashboardList/MetricsDashboardList.type'
 import { getSelectedDashboards } from '../../GCOMetricsHealthSource.utils'
 
-export function SelectGCODashboards(): JSX.Element {
+export interface SelectGCODashboardsInterface {
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/components/SelectGCODashboards/SelectGCODashboards.tsx:16]: move this to type file

File: src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.tsx (hunk 1 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.tsx:321]: this is passed as prop to a button. In button component if onClick  function is async it shows a loading spinner in button  and waits for it to resolve

File: src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/components/SelectGCODashboards/SelectGCODashboards.tsx (hunk 3 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/components/SelectGCODashboards/SelectGCODashboards.tsx:16]: done

File: src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.utils.ts (hunk 2 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.utils.ts:78]: done

File: src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.tsx (hunk 1 above)
Comment (praneshg239) [reply src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.tsx:321]: To show spinner we can depend on state right, why we need `async`?

Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.tsx]: done

**Review by praneshg239 — APPROVED** [review]

File: src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.utils.ts [hunk 4]
```diff
@@ -154,16 +159,19 @@ export function transformGCOMetricSetupSourceToGCOHealthSource(setupSource: GCOM
     if (metricInfo.higherBaselineDeviation) {
       thresholdTypes.push('ACT_WHEN_HIGHER')
     }
-
+    const regexExpression = /^\{/
     const spec: StackdriverMetricHealthSourceSpec = healthSource.spec || []
+    const isFixed = getMultiTypeFromValue(metricInfo.query) === MultiTypeInputType.FIXED
+    const startWithCurlyBraces = regexExpression.test(defaultTo(metricInfo.query?.trim(), ''))
+    const shouldParseQuery = startWithCurlyBraces ? true : isFixed
```
Comment (amitsh14) [inline src/modules/85-cv/pages/health-source/connectors/GCOMetricsHealthSource/GCOMetricsHealthSource.utils.ts:166]: Nit: const shouldParseQuery = startWithCurlyBraces || isFixed would be easy to read

**Review by amitsh14 — APPROVED** [review]

**Comment by deepesh-ui** [comment]
retrigger jest

**Pull request merged**